lib/links/parser.rb

```diff
@@ -0,0 +1,65 @@
+require 'httparty'
+
+class Parser
+
+  @@elements = { a: ['href'],
+                img: ['src'],
+                script: ['src'],
+                frame: ['src'],
+                iframe: ['src'],
+                link: ['href'],
+                source: ['src', 'srcset'],
+                track: ['src'] }
+
+  attr_reader :url
+  #attr_accessor :doc
+
+  def initialize(url)
+    @url = url
+  end
+
+  def run
+    all_links = []
+    @doc ||= get_html
+
+    @@elements.each do |elem, attr_list|
+      elems = get_elems(elem)
+
+      attr_list.each do |attr|
```

Comment:
I know I said in a previous comment you could do it with `inject`, but I actually think you should look into [`Enumerable#flat_map`](https://apidock.com/ruby/Enumerable/flat_map)

Reply:
Yes, I'm still looking into this one. But ok, I'll have a look at that one too